### DISCUSSION TITLE
Ability to Set Dimensions in the Embedding Request

### DISCUSSION THREAD

**Discussion by jmurkothintel** [microsoft/semantic-kernel#7602] [Q&A] [answered]
Right now there is no way to  pass dimensions as a part of the GenerateEmbeddingsAsync method in Semantic Kernel because the signature does not allow it. The only way seems to be  to  define the dimension at the time of adding the service to kernel.  We would definitely like to have the ability to set the dimension as part of the individual request  as a part of executionsettings similar to the pattern followed in the TextGeneration and Chat signatures. Is this functionality planned as a part of the embedding  re design? Also is there an interim way to accomplish this without having to write a custom connector?

**Comment by moonbox3** [comment, marked as answer]
Hi @jmurkothintel, thanks for your comment/question. We're working on new memory abstractions which allows for a much more custom approach to dealing with memory and embeddings. Are you targeting Python or dotnet (my guess is dotnet based on the method name you mentioned above)?

For Python you can view it here: https://github.com/microsoft/semantic-kernel/blob/feature-python-memory/python/samples/concepts/memory/new_memory.py

For dotnet you can view it here: https://github.com/microsoft/semantic-kernel/blob/feature-memory-service/dotnet/samples/Concepts/Memory/VectorStore_DataIngestion_Simple.cs#L82

**Reply by jmurkothintel** [reply]
Hi @moonbox3  I am targeting dotnet. In the example that you shared above  and in the branch I still see the same behavior ie dimension cannot be specified at the time of request in the GenerateEmbeddingAsync method which is what I was looking to do. 
![image](https://github.com/user-attachments/assets/9e1b4f0c-1713-40b2-9f3c-1546509c95c4)

**Reply by moonbox3** [reply]
Tagging @westey-m for visibility and further help.

**Reply by moonbox3** [reply]
From what I see, one defines the embedding vector dimension as part of the Glossary class:

```c#
private sealed class Glossary
{
    ...

    [VectorStoreRecordVector(1536)]
    public ReadOnlyMemory<float> DefinitionEmbedding { get; set; }
}
```

and then that is used as part of:

```c#
var glossaryEntries = CreateGlossaryEntries().ToList();
var tasks = glossaryEntries.Select(entry => Task.Run(async () =>
{
    entry.DefinitionEmbedding = await textEmbeddingGenerationService.GenerateEmbeddingAsync(entry.Definition);
}));
await Task.WhenAll(tasks);
```

There isn't an explicit argument for passing in the dimension. I will let @westey-m follow up as he's leading the design for dotnet.
